### ISSUE TITLE
Deployment template parse failed: 'Error reading string. Unexpected token: StartArray. Path '[0]'

### ISSUE THREAD

**Issue by chewong** [Azure/aks-engine#3910]
**Describe the bug**

Wasn't able to deploy the following API model:

```json
{
    "apiVersion": "vlabs",
    "location": "eastus2",
    "properties": {
        "orchestratorProfile": {
            "orchestratorType": "Kubernetes",
            "orchestratorRelease": "1.16",
            "kubernetesConfig": {
                "networkPlugin": "azure",
                "containerRuntime": "containerd",
                "useCloudControllerManager": true,
                "cloudProviderRateLimitQPS": 6,
                "cloudProviderRateLimitBucket": 20,
                "controllerManagerConfig": {
                    "--feature-gates": "CSIInlineVolume=true,LocalStorageCapacityIsolation=true,ServiceNodeExclusion=true"
                },
                "apiServerConfig": {
                    "--feature-gates": "VolumeSnapshotDataSource=true",
                    "--enable-admission-plugins": "NamespaceLifecycle,LimitRanger,ServiceAccount,DefaultStorageClass,ResourceQuota,AlwaysPullImages"
                },
                "outboundRuleIdleTimeoutInMinutes": 30,
                "addons": [
                    {
                        "name": "azuredisk-csi-driver",
                        "enabled": false
                    },
                    {
                        "name": "azurefile-csi-driver",
                        "enabled": false
                    }
                ]
            }
        },
        "masterProfile": {
            "count": 1,
            "dnsPrefix": "{dnsPrefix}",
            "vmSize": "Standard_DS2_v2",
            "availabilityProfile": "VirtualMachineScaleSets",
            "availabilityZones": [
                "1",
                "2"
            ]
        },
        "agentPoolProfiles": [
            {
                "name": "agentpool",
                "count": 2,
                "vmSize": "Standard_DS2_v2",
                "availabilityProfile": "VirtualMachineScaleSets",
                "availabilityZones": [
                    "1",
                    "2"
                ]
            }
        ],
        "linuxProfile": {
            "adminUsername": "azureuser",
            "ssh": {
                "publicKeys": [
                    {
                        "keyData": "{keyData}"
                    }
                ]
            }
        },
        "servicePrincipalProfile": {
            "clientID": "{clientID}",
            "secret": "{secret}"
        }
    }
}
```

with the following error:
```log
StatusCode=400 -- Original Error: Code="InvalidTemplate" Message="Deployment template parse failed: 'Error reading string. Unexpected token: StartArray. Path '[0]'.'."
```

**Steps To Reproduce**

**Expected behavior**

**AKS Engine version**

v0.56.0. It works on v0.55.1 though.

**Kubernetes version**

**Additional context**

**Comment by jackfrancis** [comment]
Can you try without the custom apiserver and controller-manager config to see if that's where to investigate?

**Comment by chewong** [comment]
Still getting the same error if I removed custom apiserver and controller-manager config. It works if I removed theavailabilityZones field.

**Comment by jackfrancis** [comment]
hrm

remove from which one? control plane or node config?

**Comment by chewong** [comment]
Removing master availaibiltyZones would make it work again. I think https://github.com/Azure/aks-engine/pull/3767 causes this regression. 

It looks like the master VMSS ARM template is referencing the entire string array for `zones` instead of individual values within the array:

```
...
      "sku": {
        "capacity": 3,
        "name": "[parameters('masterVMSize')]",
        "tier": "Standard"
      },
      "tags": {
        "aksEngineVersion": "[parameters('aksEngineVersion')]",
        "creationSource": "[concat(parameters('generatorCode'), '-', variables('masterVMNamePrefix'), 'vmss')]",
        "orchestrator": "[variables('orchestratorNameVersionTag')]",
        "poolName": "master",
        "resourceNameSuffix": "[parameters('nameSuffix')]"
      },
      "type": "Microsoft.Compute/virtualMachineScaleSets",
      "zones": [
        "[parameters('availabilityZones')]"
      ]
    }
...
```

**Comment by jackfrancis** [comment]
@chewong thank you for the diagnosis! fixed in #3917
